### PR TITLE
[TEP-0048] Add Proposed design details

### PR DESCRIPTION
**Pull request by vinamra28**
This proposes the design as how the `Results` will look like when we
want to declare the `optional` value for them.

Signed-off-by: vinamra28 <jvinamra776@gmail.com>

### CODE REVIEW

File: teps/0048-task-results-without-results.md
Comment (jerop) [inline teps/0048-task-results-without-results.md:465]: @ywluogg @chuangw6 @yaoxiaoqi - would appreciate your review on the default value for results

File: teps/0048-task-results-without-results.md
Comment (jerop) [inline teps/0048-task-results-without-results.md:526]: can imagine use cases where users could need this at runtime, but the use cases we have defines at at authoring time so agreed with solving for that only in this TEP

**Review by jerop — APPROVED** [review]
Thank you @vinamra28! 

Could you please show how the solution solves the [use cases](https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md#use-cases)? It's implied but it's better when we're explicit about it. For the first example, `build-image` could have the `Result` named `image-name` default to empty string `""` which can be used in the `when` expressions in the subsequent `Tasks`.

/approve

**Comment by jerop** [comment]
@pritidesai @tektoncd/core-maintainers looking for reviews here 🙏🏾

File: teps/0048-task-results-without-results.md
Comment (pritidesai) [inline teps/0048-task-results-without-results.md:8]: almost two years 😲 😿

File: teps/0048-task-results-without-results.md
Comment (pritidesai) [inline teps/0048-task-results-without-results.md:501]: I don't think the `results` can be specified in `pipelineTask`. Is this being introduced here in this proposal in addition to the default specified in `task` specification?

Comment (vinamra28) [reply teps/0048-task-results-without-results.md:501]: @pritidesai actually no 😅. In the use-cases mentioned [above](https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md#consuming-task-results-from-the-conditional-tasks), I saw this syntax, so thought that at the time of writing this TEP the author might have just displayed that the following result would be produced from the `Task`. Happy to change this  😄

Comment (pritidesai) [reply teps/0048-task-results-without-results.md:501]: oh no, I have completely forgotten what I authored 😞 

yeah if I can recollect, it's included just for the readability 😥 

let's drop `results` from the `pipelineTask` since its not something supported or will be as part of this proposal, otherwise looks good to me.

**Comment by pritidesai** [comment]
thank you @vinamra28 for working on this 👍 Please drop the `results` from `pipelineTask` before merging.

Also, we might have to specify in the TEP that this feature will available as `alpha` and make it `implementable`

/approve

**Comment by jerop** [comment]
thanks @vinamra28 🎉 

/assign @pritidesai 
/lgtm

(merging offline as it meets requirements)

**Pull request merged**
